### DISCUSSION TITLE
Pay Card

### DISCUSSION THREAD

**Discussion by annefranca** [Telefonica/mistica-design#1356] [Proposals]
### Goal

Avaliar a proposta feita no Playroom onde criamos atomicamente o Card Pay para aplicar no fluxo de Vivo Pay. Para essa proposta usamos o Snap Card como base.

[Link Playroom](https://mistica-web.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8AhCAPABABwIZSgEsA7AcwF5gAOAXwD4AdYzTVDGKTACxkNK4AulAEwBGYTUwB3QlAFdKAZgAsi%2BkxYskAZWK5sAYVwAnKBs0tCAZwCSxAG4xjVmOYuFIxSm4us7AGxIYTCs8MBhyBhAAIxgBKRgYYijGZl8LJBtPAEEwfBgAWw8AJRhSAFd-EzokAHosiGJc-KKwUoqq41T0jIamvNhW9srqur7mwZKyka6fDPriQOIYbt8aOcwYdAFjXG809J0BXDAAa1We1jQsPAISUjQBAQgCymUABnUDy9YAFS2BABWOgABSqAhI5QKmCMVi4KBOpzq-22wI282uFx6bHQWMumUWQRCYQiUVi8USyRAeJ%2BR0RNJ%2BfwBijooneAFJMLAYbg4dFEcjmQzaSiBMJMJB-BBjJRJdKrAA6AQA7QwTxQEwATzsjmcMHohQlL2wuysmD0mAAaoR7BBBdthMLDrVtMczk7fHSzsSTqSQCYIOViGZqahyk9GiDjIQClqQrH-P5MJHjDArFYPeltBBAmBCMdjOjsbUUOHnsQozGtTUXW7zkX5gEgh66pii3VXfSNutvrVVq32FBUgPcUwmDicPgiGRKLR%2B9cONxePwhMBRKI1NJZPIRO9PlidHpDCYzN9LLYHE4XBsPI19vim8sfeFIjE4gkkikG34cgNClMOlGepfxaACZhbcY-yGaZOhrSDQLaGDqnbBYlhWbsNgBXZ72xTt3W-JBRWBQAiAlI4jMHXcR7SBYUOzrGkewsPtzBHDhhxLDBUhAAAaEB5EKNMEAAbRAa1bQAfQAORgKQQAAXV4mQ5C4KxhMUAA2d45JoIA)

### Context

No fluxo de Pay, necessitamos de um componente (Large/Small) para suprir à demanda de aplicação como mostra o arquivo do Figma anexado abaixo.

Gostaríamos de entender se é possível a aplicação do background com gradiente, como mostra no arquivo. 

Observação: não temos a dependência de CMS em Pay.

### Impact

Adicionar esse componente na nossa biblioteca reduziria o tempo de desenvolvimento e ajudaria no processo de criação dentro das squads. 

### Discussion Questions

ND

### Do you have a visual proposal?

https://www.figma.com/file/YbfTZMCjr75WIkPjyxmDDT/Vivo-Card-Pay?type=design&node-id=0%3A1&mode=design&t=Yh2xqF6GCrC1DlMn-1

**Comment by yceballost** [comment]
Did you try DisplayDataCard for this purpose? 
https://tinyurl.com/2xmzec8w

It is very similar and you avoid generating a new component

**Reply by yceballost** [reply]
At the same time I think could be interesting to have a "CreditCard" with custom proportions and custom backgrounds for Credit card uses. I wouldn't know how much of a core component that component could be, we would have to look at the current uses of other Telefónica products to see if it could cover all cases.

**Reply by annefranca** [reply]
E o BG gradiente, conseguimos chegar nesse resultado usando o Display Data Card?

**Reply by annefranca** [reply]
Não consegui chegar nesse resultado do Playroom pelo Figma. 
Outra dúvida: teria a opção de não deixar o ícone da direita clicável?

**Reply by yceballost** [reply]
> E o BG gradiente, conseguimos chegar nesse resultado usando o Display Data Card?

If we evolve the DisplayDataCard to support custom background colour, it could support gradients. Ideally, we would then create a colour token for gradients.

> Outra dúvida: teria a opção de não deixar o ícone da direita clicável?

no, currently that wouldn't be possible, but I can think of this hack 🤔
https://tinyurl.com/2977ggro

We can check with developers if would be possible (I think yes)

**Reply by annefranca** [reply]
@yceballost perfeito! Podemos seguir com essa solução para o "Card Large". Essa solução apresentada cobre o "Card Small"? https://www.figma.com/file/YbfTZMCjr75WIkPjyxmDDT/Card-Pay?type=design&node-id=1502%3A25235&mode=design&t=K6kwNN71coGnOC7z-1

**Reply by annefranca** [reply]
@aweell @AnaMontes11 
Segue a proposta para aplicação do BG em gradiente: [Playroom](https://mistica-web.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8BnALgTwDYwHwB0A7AAmOAAMijTSA6AIwEMwBrAcwCcIBXQqAYUYcoARjLUaxJq048%2B8YlgCWhGEIC0nRlCUxCaABQiATAAZYbADTEAxADY7p0wE5n1mwBY7IgKIBWADEASgBuCVIAX3DiaIZmdi5eASEoY3ESSWkEuSgFZVUNLR09QxNzGCtbBydXdyd601DoqIzSWKzZJMFhAGZ0ySl4zvlFFTUOTQ5tXX0jMwt3apc3Wxh6HrAAdigm1uIW6KoMiPIWpAB6dGw8I6QAcQ4lKGIuAHcUAF5gEQjiSCxuABbQifYA9CIEDL3R5QACSaBggMhkmiSAAQhAAB7EAAO2h0hDYXwAHBDoqR0VjcfiVGwAJokiFIAAqMExaA8uH4EA4MGIUEYxEBQjAjGsOjYSjQjCw1hgAEduEplPQplAIBdWezORcMZjkQNKZiYM9Xk80AALL49TZ%2BX5KFCwwgANxgHBQfLAWEYKBQADlGICYB98CAOok%2BN1RKGDQNiEbqVACUTvnZfgLpQBBNBoR70bgI0GhgASalgHFDZL2caQAGVpaxiCg8WBg8BTFW4zWnfk%2Bc3mMHQ-QYGhXjA9DHyV2KbDIIRM6LYIClGAAEoVbjeji4C6ziDzxeIlfrtiboSx6c0JB7g-aI9rjdbnfnG8Lu-Lh%2Bnp9Tmsvwi9i9pwTPEk1pDEcwgQEvg8DtE2TBlgBgztL0kFk2TQPxcAABW9NAVCBYhBBQC00XiTV0Mwn9DXOPVAK7I06KAnsxibFtBzDEcxwnEBGMvOsGxYViBy%2BDteJQtD2R6XARFMABSfk%2BSIi0snIySxL4rU0DSf4eS%2BHT3VoBF2VrGA5wFDgMCdV13RgZCUOnRE-kgnEphQYhCEFAA1JRnQgKjxPOTTjHU3963iELDTCxt%2B1bEMQCEHJJ2reyKTRAs0H3LDHmFCym2FLAsGITLeV9CLp1rCBlDAKUhH8viaPS4qlByjAyqvc4opYNrd3-MYIt1LF1IuTreIG-UpzGk0LzGyFUT1ODaUZOijRNYgzSgS0vjMWCHSst0PT%2Bb1fQDIM4vDHIoxEJKgPmkDky%2BEQ035RgsxzPMCxgIsQFLO8KxAOz6M6oTYrBUS6pnXrVGB9jh1HcdCGulL41fQ8PxPM9t13Oc3yXY9H3PcGrxR988a-c8sf3HH73R79kt-ZjVH6zTMMAIgI2ZZ4gRC54xVIwoaOoE0aaMGibheNKBprFg0LgeJ54URaXZbhBEkTmqk7sW4BSWW26aUJBDtYkjlsN5ZcgWsFQERBCA5UxHE3TQXmdSlia9VW9bNrBW17UdF19s9I7-UDGGhgjZJhGMRGa110DCQep6M0YbNcyUfNCy%2BEsyzdSshqBmK2zBunDQZvs2LOzj4aj%2Bzr2x1HSYx59idxz8G8Jmda5Jlunwp29m5pgmi9Qv8AMJ4C9bYcCMqgxDYI1-XoMLpH42Z7DcPwwFCJ9EiyMCij%2Bpdwer1otvkch0vhKHCvuLaik87L9sAaXo2pJk%2BTYE34iVN3tST9QoKnKwLpYA%2BkUCGXQiZMyQhLJ%2Bxso-JexBHKQEBC5H07kvI%2BT8ofIC38tI33jALcKv94x3wvvFMOVcn5pQgoQLKzUoF5RlIVYqn0UB4MkBVKqNUKxYPog1ahtCWp4OGoLE%2BPUR48JmnVYRhCi6SOSpNCWqID5q2xHPFM2tXZYnduaK0wBto%2Bz2jZQ6Pog6nSHKHC6KRI48SomPWOKZHrpheknN6qcPpfR%2BuWHOUiSEgx6IvauJdoblzhtfURTdqb40xi%2BDufcondQiWjeJPcqZJLJtuUew8%2BqjxXuzDmXMTC80oofaRrAhbH3kWLKaSiKkUnOEreWqsoQNJVtNGO90tZwLsfBJaRtOSCEAVwFANtiDOlTlMfQMBxRKElNKLATtny1MNG7U0OjrTe2ILtGBB0vQmJOiHGQYcow9AoUfdW4945ONeinNOn0M7fSzn9OBqFfEF2ecXM%2BwTL6hIRjYnh7dKZ1y7uTGJgLO79wyf85GsTInpMbjCtJrcJFZMZqPdpYEIAQWnkhBa88Z7vJrCvHCL114f23qwIp%2B8lniWpYEz5%2BcQlcV%2BUI15IkCUaXQi-OSCkyVf00lJIhFJ-76T0pVHkoCjJoAgfucy0DrIenZShRBzlXJoOIN5Xygr8FBSEQQspRD%2BLxC%2BWQxKfz4FH0ajQ7K9CUD5SYValhbCaAcJXFwrVupLUCKgbqkaoiUU3GRbS0KIjZEH1-CsyWFSVG4vUV0lZa01l6KcAY7ZAc9nBzOhYroKQTlmvouiuOqYrkuJue4%2B5njs7-VzgJYJoNFWn17Ma2GTLTlMQRfXbuoLe6wqRU-RJHaQX9uBZClKYjskSNyezTm3NKVSL1V1WxYbqIRpqSLKE9SYSNMhDLGEkIQCWBAJaREn0EAAG0QAaogAAfT9DAV4IAAC6B6PYWhQGekQPQ7DEgfREIAA)

**Reply by aweell** [reply]
> Essa solução apresentada cobre o "Card Small"?

The card small could be created using an snapCard, there's already a solution linked https://tinyurl.com/2a9db4de in the Figma that would work.

Although can this pay card component can be created using core components there are some drawbacks I suggest to think about:

* If the definition of the core components changes, this PayCard will be affected
* We need to allow, in the same way we did with the poster card, the possibility of having a custom background in the snapCard and the displayMediaCard, so a solid or a gradient color can be applied via that prop. In that case no matter where the gradient definition comes, from the skin or just custom.
* The snapCard proposal is not using any text properties of the component, instead, is using the extra to provide the information, I'm wondering if it's worth it to use a component just to place the top icon.
* This is a highly specialised component, that means that only serves one purpose, show a user a payment card information / payment related actions.
* If the requirements change in the future, I'm thinking about, owner information, CVV, due date (or other related payment information) it will be probably be added in the extra, ending in the same scenario we're right now.

The conclusion here is, we have two roads:

* Update two Mística core components to allow custom background color, place most of the info in the extra and be aware that future system requirements could lead to visual / or development breaking changes.
* Create an extended component

**Reply by BrunoBiagioniNeto** [reply]
@aweell  já falamos disso e ter "hacks" para funcionar pode não ser bom mesmo, no futuro pode haver quebra de layout com alguma atualização.

Podemos propor um novo card de cartão de crédito com informações que você falou para ser Mística se acharem que pode ser Global, caso contrário iremos para Extended.

[Propostas de Templates A e B](https://tinyl.io/91x0)

[Card Promo](https://tinyl.io/91x9)

Seguimos com Mística?
ou
Seguimos com Extended Vivo?

**Reply by aweell** [reply]
Extended is the way since ATM there's not a clear list of global requirements for this component.
